### PR TITLE
WIP: Mat3x4

### PR DESCRIPTION
**Pull request by emilk**
Part of https://github.com/bitshifter/glam-rs/issues/25

This was more work than I thought (it always is). I can't say I fully grok how all the different traits fit together, but I tried to copy the basic structure of the 4x4 types. I still have some hard-to-decipher errors.

## TODO:

* [ ] Get it to compile
* [ ] Finish up the TODO:s an commented code
* [ ] Write tests
* [ ] Write benchmarks

### CODE REVIEW

**Comment by bitshifter** [comment]
Thanks for that, I haven't had a chance to have a proper look yet, hopefully I'll find some time over the long weekend. It might pay to flesh things using the public types and look at the core stuff later. It is a bit complicated unfortunately and I haven't had to think about how to deal with new types yet.

**Comment by emilk** [comment]
No hurry - I (probably) won't continue work on this until Tuesday. And yeah, I should have started with

``` rust
struct Mat3x4 {
    x_row: Vec4,
    y_row: Vec4,
    z_row: Vec4,
}
```

I might make another PR with that so we can try that out and do benchmarks etc.

File: src/mat3x4.rs [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+    ($t:ident, $vec4:ident, $vec3:ident, $quat:ident, $inner:ident) => {
+        /// A 3x4 matrix with all elements set to `0.0`.
+        pub const ZERO: Self = Self($inner::ZERO);
+
+        /// A 3x4 identity matrix, where all diagonal elements are `1`, and all off-diagonal elements are `0`.
+        pub const IDENTITY: Self = Self($inner::IDENTITY);
+
+        /// Creates a 3x4 matrix from threw row vectors.
+        #[inline(always)]
+        pub fn from_rows(x_row: $vec4, y_row: $vec4, z_row: $vec4) -> Self {
```
Comment (bitshifter) [inline src/mat3x4.rs:63]: These should be `$vec3`

**Comment by emilk** [comment]
Closed in favor of https://github.com/bitshifter/glam-rs/pull/157

**Pull request closed without merging**